### PR TITLE
[Bugfix] permissions center bug fixes

### PR DESCRIPTION
**Pull request by thenamankumar**
closes: https://github.com/sourcegraph/sourcegraph/issues/49168
closes: https://github.com/sourcegraph/sourcegraph/issues/49285

- Allow non admin users to view their permissions sync jobs
- Allow site admins on dotcom to view user's email via `User` gql type. They already can view that from `SiteUser` gql type used in user management.

## Test plan

- added unit tests

### CODE REVIEW

**Review by sashaostrikov — COMMENTED** [review]
LGTM!
Make sure to add a `backport 5.0` label to the PR before the merge and I'll approve it 👍

**Review by sashaostrikov — APPROVED** [review]

File: cmd/frontend/graphqlbackend/user.go [hunk 1]
```diff
@@ -111,17 +111,9 @@ func (r *UserResolver) DatabaseID() int32 { return r.user.ID }
...
-			return "", err
-		}
-	} else {
-		// 🚨 SECURITY: Only the user and admins are allowed to access the email address.
-		if err := auth.CheckSiteAdminOrSameUser(ctx, r.db, r.user.ID); err != nil {
-			return "", err
-		}
+	// 🚨 SECURITY: Only the user and admins are allowed to access the email address.
+	if err := auth.CheckSiteAdminOrSameUser(ctx, r.db, r.user.ID); err != nil {
+		return "", err
```
Comment (pjlast) [inline cmd/frontend/graphqlbackend/user.go:116]: Are we sure about this one? This seems like a non-trivial change for Sourcegraph.com, I'd run it over with Security @sourcegraph/security

File: cmd/frontend/graphqlbackend/user_test.go [hunk 2]
```diff
@@ -146,9 +146,13 @@ func TestUser_Email(t *testing.T) {
 		envvar.MockSourcegraphDotComMode(true)
 		defer envvar.MockSourcegraphDotComMode(orig)
 
+		users := database.NewMockUserStore()
```
Comment (sashaostrikov) [inline cmd/frontend/graphqlbackend/user_test.go:149]: let's add a couple of more test cases, we currently have only a negative test

File: cmd/frontend/graphqlbackend/user.go (hunk 1 above)
Comment (kopancek) [reply cmd/frontend/graphqlbackend/user.go:116]: Why do we need the email on the permissions screen anyway? Would it be enough to remove it from the graphql request?

Comment (evict) [reply cmd/frontend/graphqlbackend/user.go:116]: I don't necessary see a problem with this change. It's already possible for site-admins to view user emails through the GraphQL API.

Comment (thenamankumar) [reply cmd/frontend/graphqlbackend/user.go:116]: As @evict said, it is already possible on dotcom for admins to see emails via `SiteUser.email` GQL field in User Management.

This PR brings security checks consistency between `User.email` and `SiteUser.email`.

**Review by sashaostrikov — APPROVED** [review]
LGTM!

**Pull request merged**